base/cvd/cuttlefish/host/libs/zip/lazily_loaded_file.cc

```diff
@@ -152,16 +165,37 @@ Result<size_t> LazilyLoadedFile::Impl::Read(char* data, size_t size) {
     size_t read_request = std::min(*end_of_present_data - seek_pos_, size);
     ssize_t data_read = contents_file_->Read(data, read_request);
     CF_EXPECT_GE(data_read, 0, contents_file_->StrError());
+    LOG(VERBOSE) << "Read " << data_read << " from local storage, seek pos was "
+                 << seek_pos_;
     seek_pos_ += data_read;
     return data_read;
   }
   CF_EXPECT(callback_->Seek(seek_pos_));
-  size_t data_read = CF_EXPECT(callback_->Read(data, size));
-  CF_EXPECT_EQ(WriteAll(contents_file_, data, data_read), data_read,
-               contents_file_->StrError());
-  already_downloaded_.InsertRange(seek_pos_, seek_pos_ + data_read);
-  seek_pos_ += data_read;
-  return data_read;
+  if (size < kMinReadSize && seek_pos_ + kMinReadSize < size_) {
```

Comment:
This only extends the size if it can be extended to at least `kMinReadSize`. Consider extending every time `size < kMinReadSize` to `min(kMinReadSize, size_ -seek_pos_)`

Reply:
Good point, fixed.